### PR TITLE
Balance caching microservice

### LINKED ISSUE #281: Add functionality to balance caching service
- [x] mock data for graph creation on front-end.
- [x] remove GET endpoint and return the data in portfolio instead. 
- [x] add source balances. Needed for the graph in portfolio page.
- [x] stop auto logging if user not logged for more than 3 months
- [x] DateTimeFields errors and fix DB save logic

### PR DESCRIPTION
**Pull request by KoliosterNikolayIliev**
#### Front-end needs to be refactored in order to handle the additional data returned from Portfolio !!!
 - file - cash_balance.py, row: 22

### Control environment variables:

- DB_AUTO_CACHING_INTERVAL=1440 - controls auto update interval (minutes)
- USE_BALANCE_CACHE_MOCK=False - balance caching returns mock data if True
- AUTO_UPDATE_USER_DELAY=0 - controls auto update for the next user(seconds)

data.env - pinned to slack.

closes #281

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -50,25 +50,25 @@ services:
...
+#
+#    ports:
+#      - "5001:5001"
+#
+#    environment:
+#      - FLASK_APP=portfolio
+#      - FLASK_ENV=development
+#
+#    env_file:
+#      - data.env
```
Comment (Sheko1) [inline docker-compose.yml:71]: please revert this changes

File: docker-compose.yml [hunk 2]
```diff
@@ -106,18 +106,19 @@ services:
...
+#      - manage.py
+#      - runserver
+#      - --noreload
+#      - 0.0.0.0:8009
+#
+#    ports:
+#      - "8009:8009"
+#
+#    env_file:
+#      - data.env
```
Comment (Sheko1) [inline docker-compose.yml:124]: same as above

File: server/balances_history_cacher/balances_history_cacher/settings.py [hunk 3]
```diff
@@ -96,7 +95,7 @@
 DATABASES = {
     'default': {
         'ENGINE': 'djongo',
-        'NAME': 'balance_cache',
+        'NAME': 'test_balance_cache',
```
Comment (Sheko1) [inline server/balances_history_cacher/balances_history_cacher/settings.py:98]: why name of the database is **test_balance_cache**

File: docker-compose.yml (hunk 2 above)
Comment (KoliosterNikolayIliev) [reply docker-compose.yml]: Done

File: server/balances_history_cacher/balances_history_cacher/settings.py (hunk 3 above)
Comment (KoliosterNikolayIliev) [reply server/balances_history_cacher/balances_history_cacher/settings.py]: Done

**Review by DimoDimchev — APPROVED** [review]

**Review by Sheko1 — APPROVED** [review]
